pints/tests/test_toy_german_credit_hierarchical_logpdf.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+#
+# Tests the German credit toy distribution.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import pints.toy
+import unittest
+import numpy as np
+import io
+import urllib
+import urllib.request
+from scipy import stats
+
+
+class TestGermanCreditHierarchicalLogPDF(unittest.TestCase):
+    """
+    Tests the logpdf toy distribution from fitting a hierarchical logistic
+    model to German credit data.
+    """
+    @classmethod
+    def setUpClass(cls):
```

Comment:
Why are the tests downloading too?
You don't need to test if downloading works by downloading twice :D

Reply:
In fact, you can't test if the downloaded value is correct because we don't know a-priori what it's going to be.
It should be possible for them to correct errors in the data, for example.

You can only test if it's still well-formed, e.g. can be read as data in the format we expect. So array shapes and sizes, that sort of thing.

Reply:
Yep, good points; ha, was being dense before! I have now amended and added tests of shapes. The dataset and repo and old and well-used, so not anticipating any changes to it though.